File: JavaScript/Condition.js

let test1 = true;
let test2 = false;

var log = "\r\nこれから条件分岐を試します\r\n\r\n";

if (test1) {
	log+="どうやらtest1は真のようです\r\n";
}

if (!test1) log+="どうやらtest1は偽のようです\r\n";
else if (test2) log+="どうやらtest2は真のようです\r\n";
else if (!test2 && test1) log+="真偽が混在しているようです\r\n";
else log+="どれでもないようです\r\n";


let val = 8;
switch (val) {
	case 0:
	case 1:
		log+=val+"は特別な値である\r\n";break;
	case 8:
	case 9:
	case 10:
		log+="これに隣接する数にも言えたりするが,\r\n";
		// breakを省くと,case 8,9,10を実行すれば,直下のcase 4,6も実行する
	case 4:
	case 6:
		log+=val+"は素数ではない\r\n";break;
	case 2:
	case 3:
	case 5:
	case 7:
		log+=val+"は素数である\r\n";break;
	default:
		log+=val+"なぞ私には興味がない\r\n";break;
}

console.log(log);

File: Go/Condition.go
package main

import "fmt"

func Condition() {

	test1, test2 := true, false

	fmt.Println("\r\nこれから条件分岐を試します\r\n")

	if (test1) {
		fmt.Println("どうやらtest1は真のようです")
	}

	if (!test1) {
		fmt.Println("どうやらtest1は偽のようです")
	} else if (test2) {
		fmt.Println("どうやらtest2は真のようです")
	} else if (!test2 && test1) {
		fmt.Println("真偽が混在しているようです")
	} else {
		fmt.Println("どれでもないようです")
	}

	var val int = 8

	switch val {
		case 0,1:
			fmt.Println(val,"は特別な値である")
		case 8,9,10:
			fmt.Print("これに隣接する数にも言えたりするが,")
			fallthrough
			// fallthrough により,case 8,9,10を実行すれば,直下のcase 4,6も実行する
		case 4,6:
			fmt.Println(val,"は素数ではない")
		case 2,3,5,7:
			fmt.Println(val,"は素数である")
		default:
			fmt.Println(val,"なぞ私には興味がない")
			switch {
				case val < 0 :
					fmt.Println("なにせ,",val,"は小さすぎるのだ")
				case val > 10 :
					fmt.Println("なにせ,",val,"は大きすぎるのだ")
				case (val > 0)||(val < 10) :
					fmt.Println("なにせ,",val,"は中途半端なのだ")
			}
	}

	fmt.Println("\r\n\r\n")

}

/*
	Go では次のように if 節の終了直後に else を置かなければならないみたい
	elseの直前に改行があってはいけない
	if {
		// something
	} else {
		// something
	}
*/